Realm/Tests/RLMMultiProcessTestCase.m

```diff
@@ -0,0 +1,119 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMMultiProcessTestCase.h"
+
+@interface RLMMultiProcessTestCase ()
+@property (nonatomic) bool isParent;
+@property (nonatomic, strong) NSString *testName;
+
+@property (nonatomic, strong) NSString *xctestPath;
+@property (nonatomic, strong) NSString *testsPath;
+@end
+
+@implementation RLMMultiProcessTestCase
+// Override all of the methods for creating a XCTestCase object to capture the current test name
++ (id)testCaseWithInvocation:(NSInvocation *)invocation {
+    RLMMultiProcessTestCase *testCase = [super testCaseWithInvocation:invocation];
+    testCase.testName = NSStringFromSelector(invocation.selector);
+    return testCase;
+}
+
+- (id)initWithInvocation:(NSInvocation *)invocation {
+    self = [super initWithInvocation:invocation];
+    if (self) {
+        self.testName = NSStringFromSelector(invocation.selector);
+    }
+    return self;
+}
+
++ (id)testCaseWithSelector:(SEL)selector {
+    RLMMultiProcessTestCase *testCase = [super testCaseWithSelector:selector];
+    testCase.testName = NSStringFromSelector(selector);
+    return testCase;
+}
+
+- (id)initWithSelector:(SEL)selector {
+    self = [super initWithSelector:selector];
+    if (self) {
+        self.testName = NSStringFromSelector(selector);
+    }
+    return self;
+}
+
+- (void)setUp {
+    self.isParent = !getenv("RLMProcessIsChild");
+
+    NSProcessInfo *info = NSProcessInfo.processInfo;
+    self.xctestPath = info.arguments[0];
+    self.testsPath = [info.arguments lastObject];
+
+    [super setUp];
+}
+
+- (void)deleteFiles {
+    // Only the parent should delete files in setUp/tearDown
+    if (self.isParent) {
+        [super deleteFiles];
+    }
+}
+
+- (int)runChildAndWait {
+    NSString *testName = [NSString stringWithFormat:@"%@/%@", self.className, self.testName];
+    NSMutableDictionary *env = [NSProcessInfo.processInfo.environment mutableCopy];
+    env[@"RLMProcessIsChild"] = @"true";
+
+    NSPipe *outputPipe = [NSPipe pipe];
+    NSFileHandle *handle = outputPipe.fileHandleForReading;
+
+    NSTask *task = [[NSTask alloc] init];
+    task.launchPath = self.xctestPath;
+    task.arguments = @[@"-XCTest", testName, self.testsPath];
+    task.environment = env;
+    task.standardError = outputPipe;
+    [task launch];
+
+    NSFileHandle *err = [NSFileHandle fileHandleWithStandardError];
+    NSData *delimiter = [@"\n" dataUsingEncoding:NSUTF8StringEncoding];
+    NSMutableData *buffer = [NSMutableData data];
+
+    // Filter the output from the child process to reduce xctest noise
+    while (true) {
+        NSUInteger newline = [buffer rangeOfData:delimiter options:0 range:NSMakeRange(0, buffer.length)].location;
+        if (newline != NSNotFound) {
```

Comment:
The child process writing something to stderr that doesn't have a trailing newline would be weird, but not an error.

Now that I look at it again the code is wrong though (it needs to loop until there's no newlines left since `availableData` may return something with multiple lines...).